### DISCUSSION TITLE
Bricked device recovery

### DISCUSSION THREAD

**Discussion by gabonator** [gabonator/LA104#47] [General]
Dont panic, LA104/DS203/DS213 devices are almost impossible to brick. This discussion will contain guides how to unbrick your device

| Device | Firmware upload method | Firmware location |
|  ------ | --------------------------| ------------------ |
| LA104 | [SWD interface over USB](https://github.com/gabonator/LA104/tree/master/tools/openocd) | TBD |
| DS203 HW >= 2.8x | [stm32flash](https://github.com/gabonator/LA104/tree/master/tools/stm32flash) |  load 2-5 of [RECOVERY281282.zip](https://github.com/gabonator/LA104/blob/master/tools/recovery/RECOVERY281282.zip) |
| DS203 HW <= 2.72 | [stm32flash](https://github.com/gabonator/LA104/tree/master/tools/stm32flash) |  as previous with [hwV272-V261FPGA.zip](https://github.com/gabonator/LA104/blob/master/tools/recovery/hwV272-V261FPGA.zip) |

**Comment by erik-smit** [comment]
hwV272-V261FPGA.zip only contains the FPGA. And it's in the ADR+BIN format.
https://github.com/gabonator/LA104/files/6096611/DS203HW260-272.zip contained all firmware I needed to recover my DS203.

**Comment by bjoerngiesler** [comment]
I successfully "bricked" my 213 by installing your firmware (which worked fine) and then trying to revert to the original firmware from here: http://www.minidso.com/forum.php?mod=viewthread&tid=3357&extra=page%3D1%26filter%3Dtypeid%26typeid%3D154 

Flashing works fine, but the device says "FPGA Configuration Err". I already tried the recovery image from this thread: http://www.minidso.com/forum.php?mod=viewthread&tid=4421&extra=page%3D1, and flashing works great but the error stays.

Any advice on what I can do would be highly appreciated :-)

**Comment by syproduction** [comment]
Hello. I have DS203 hw 2.81. Was bricked with FPGA error. I did STM32FLASH with dso_quad_bootloader_flash_modified.bin
Now when i connect it with DFU mode it just connects for 1 second to computer and then immediately disconnect. Going like cycle.
And i cannot flash it with STM32FLASH over ttl adapter now with "Failed to init device." error.
Is there a way to sort this out?

**Reply by gabonator** [reply]
Try different computer (had the same problem on osx)

**Reply by 203Null** [reply]
My LA104's FPGA seems to be no longer working properly. Is this something possible to fix?
